spec/requests/api/user_can_purchase_a_subscription_spec.rb

```diff
@@ -0,0 +1,133 @@
+require "stripe_mock"
+
+RSpec.describe "POST api/subscriptions" do
+  let!(:stripe_helper) { StripeMock.create_test_helper }
+  before(:each) { StripeMock.start }
+  after(:each) { StripeMock.stop }
+
+  let(:card_token) { stripe_helper.generate_card_token }
+  let(:invalid_token) { "123456789" }
+
+  let(:product) { stripe_helper.create_product }
+  let!(:plan) {
+    stripe_helper.create_plan(
+      id: "platinum_plan",
+      amount: 1000000,
+      currency: "usd",
+      interval: "month",
+      interval_count: 12,
+      name: "Berlingo News Premium Platinum Plan",
+      product: product.id,
+    )
+  }
+
+  let(:user) { create(:user) }
+  let(:user_credentials) { user.create_new_auth_token }
+  let(:headers) { { HTTP_ACCEPT: "application/json" }.merge!(user_credentials) }
+
+  describe "with valid stripe token" do
+    describe "successfully" do
+      before do
+        post "/api/subscriptions",
+             params: {
+               stripeToken: card_token,
+               email: user.email,
```

Comment:
If you use this WITH authentication, then you don't have to send off email as params, nor do you have to ask for it in the form. Just use the one that is attributed to `current_user`

Reply:
Updated this

Reply:
Looks good!